server/routerlicious/packages/services/src/documentManager.ts

```diff
@@ -5,21 +5,136 @@
 
 import { ScopeType } from "@fluidframework/protocol-definitions";
 import { BasicRestWrapper } from "@fluidframework/server-services-client";
-import { IDocumentManager, IDocument, ITenantManager } from "@fluidframework/server-services-core";
+import {
+	IDocumentManager,
+	IDocument,
+	ITenantManager,
+	IDocumentStaticProperties,
+	ICache,
+} from "@fluidframework/server-services-core";
 import { generateToken, getCorrelationId } from "@fluidframework/server-services-utils";
+import * as Redis from "ioredis";
+import { Lumberjack, getLumberBaseProperties } from "@fluidframework/server-services-telemetry";
+import { RedisCache } from "./redis";
 
 /**
  * Manager to fetch document from Alfred using the internal URL.
  */
 export class DocumentManager implements IDocumentManager {
+	private static documentStaticDataCache: ICache;
+	/** True if the cache was initialized by setStaticProperty. Becomes false if the cache is overwritten by readDocument. */
+	private static staticCacheInitializedManually: boolean;
+
 	constructor(
 		private readonly internalAlfredUrl: string,
 		private readonly tenantManager: ITenantManager,
-	) {}
+	) {
+		// TODO: Eventually replace the static documentStaticDataCache field with a passed in optional value, to not have to initialize it here
+		// This can be done once the unified RedisCache/ICache implementation is made, the issue right now is that different RedisCache objects
+		// could be passed in, which would mean different headers and therefore multiple static document caches could exist at once.
+		// The unified RedisCache implementation will have more flexibility around headers, which will help fix this issue.
+
+		// If a redis cache does not yet exist, create one to populate the static field
+		if (!DocumentManager.documentStaticDataCache) {
+			const redisConfig: Redis.RedisOptions = {
+				host: "redis",
+				port: 6379,
+			};
+			const redisClient = new Redis.default(redisConfig);
+			DocumentManager.documentStaticDataCache = new RedisCache(redisClient);
```

Comment:
Like we talked, we don't to create a detailed implementation. We should inject `ICache` from routerlicoius to here, given this is routerlicious. We should programing to abstractuct, meaning here we shouldn't see anything regarding redis, but use abstrraction `ICache`

Reply:
For this, do you recommend passing the ICache object into the constructor? The only other way I can think of is to pull the redisConfig from r11s config.json, but then there would still be the Redis.defualt and RedisCache being used here.

I was thinking before of passing the ICache into the constructor, but I plan to use this class in Historian which currently uses a different implementation of ICache/Redis cache. The Historian implementation includes a different key prefix, which would mean that we have multiple copies of the static data cache if someone creates/uses a documentManager somewhere other than historian, since the prefixes of the keys are different. This would lead to more cosmosdb calls. 

To get around this issue, I was going to wait until I made the unified RedisCache for FHL week, and allow more flexibility with the prefixes. This would allow me to go back to here, pass cache through the constructor, and then not have to worry about different prefixes creating multiple copies of the cache. 

Although since FHL starts on Monday, I can probably just inject it to the constructor already, and then do the historian PR after I do a PR for the RedisCache changes.

Reply:
Resolved in latest commit, will make other changes later depending on timeline of the RedisCache consolidation